Xamarin.Forms.Platform.Android/AppCompat/MasterDetailContainer.cs

```diff
@@ -160,5 +107,59 @@ public void SetFragmentManager(FragmentManager fragmentManager)
 			if (_fragmentManager == null)
 				_fragmentManager = fragmentManager;
 		}
+
+		void SetCurrentFragment(Fragment fragment)
+		{
+			FragmentTransaction fragmentTransaction = FragmentManager.BeginTransaction();
+
+			if (_currentFragment != null)
+				fragmentTransaction.Remove(_currentFragment);
+
+			fragmentTransaction.Add(Id, fragment);
+			fragmentTransaction.SetTransition((int)FragmentTransit.None);
+
+			// The view we're hosting in the fragment was never created (possibly we're already 		
+			// navigating to another page?) so there's nothing to commit synchronously.
+			if (_pageContainer == null)
+			{
+				// We don't currently support fragment restoration 
+				// So we don't need to worry about loss of this fragment's state
+				fragmentTransaction.CommitAllowingStateLoss();
+			}
+			else
+				CommitTransaction(fragmentTransaction);
+
+			_currentFragment = fragment;
+		}
+
+		void ClearCurrentFragment()
+		{
+			if (_currentFragment == null)
+				return;
+
+			FragmentTransaction fragmentTransaction = FragmentManager.BeginTransaction();
+			fragmentTransaction.Remove(_currentFragment);
+			fragmentTransaction.SetTransition((int)FragmentTransit.None);
+
+			// This is a removal of a fragment that's not going on the back stack; there's no reason to care
```

Comment:
This comment is now confusing here - some version of it should probably be in CommitTransaction.